com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -406,15 +574,93 @@ static IEnumerable<string> CheckOutputTensorPresence(Model model, int memory)
+        /// the model and the BrainParameters.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters
+        /// </param>
+        /// <param name="brainParameters">
+        /// The BrainParameters that are used verify the compatibility with the InferenceEngine
+        /// </param>
+        /// <param name="sensors">Attached sensors</param>
+        /// <param name="observableAttributeTotalSize">Sum of the sizes of all ObservableAttributes.</param>
+        /// <returns>A IEnumerable of the checks that failed</returns>
+        static IEnumerable<FailedCheck> CheckInputTensorShape(
+            Model model, BrainParameters brainParameters, ISensor[] sensors,
+            int observableAttributeTotalSize)
+        {
+            var failedModelChecks = new List<FailedCheck>();
+            var tensorTester =
+                new Dictionary<string, Func<BrainParameters, TensorProxy, ISensor[], int, FailedCheck>>()
+            {
+                {TensorNames.PreviousActionPlaceholder, CheckPreviousActionShape},
+                {TensorNames.RandomNormalEpsilonPlaceholder, ((bp, tensor, scs, i) => null)},
+                {TensorNames.ActionMaskPlaceholder, ((bp, tensor, scs, i) => null)},
+                {TensorNames.SequenceLengthPlaceholder, ((bp, tensor, scs, i) => null)},
+                {TensorNames.RecurrentInPlaceholder, ((bp, tensor, scs, i) => null)},
+            };
+
+            foreach (var mem in model.memories)
+            {
+                tensorTester[mem.input] = ((bp, tensor, scs, i) => null);
+            }
+
+            for (var sensorIndex = 0; sensorIndex < sensors.Length; sensorIndex++)
+            {
+                var sens = sensors[sensorIndex];
+                if (sens.GetObservationShape().Length == 3)
+                {
+                    tensorTester[TensorNames.GetObservationName(sensorIndex)] =
+                        (bp, tensor, scs, i) => CheckVisualObsShape(tensor, sens);
+                }
+                if (sens.GetObservationShape().Length == 2)
+                {
+                    tensorTester[TensorNames.GetObservationName(sensorIndex)] =
+                        (bp, tensor, scs, i) => CheckRankTwoObsShape(tensor, sens);
+                }
+                if (sens.GetObservationShape().Length == 1)
+                {
+                    tensorTester[TensorNames.GetObservationName(sensorIndex)] =
+                        (bp, tensor, scs, i) => CheckRankOneObsShape(tensor, sens);
+                }
+
+            }
+
+            // If the model expects an input but it is not in this list
+            foreach (var tensor in model.GetInputTensors())
+            {
+                if (!tensorTester.ContainsKey(tensor.name))
+                {
+                    failedModelChecks.Add(FailedCheck.Warning("Model requires an unknown input named : " + tensor.name
```

Comment:
The presence of legacy tensors names will be checked here.

Reply:
Not sure the warning message makes sense then. Should it be something like `$"Model contains an unexpected input named: {tensor.name}"`